src/main/java/spoon/support/compiler/SpoonPom.java

```diff
@@ -517,39 +516,23 @@ private void generateClassPathFile(File mvnHome, MavenLauncher.SOURCE_TYPE sourc
 	}
 
 	/**
-	 * Try to guess Maven home when none is provided.
-	 * @return the path toward maven install on the local machine.
+	 * Lookup absolute path to maven executable.
+	 * @return the path toward maven executable on machine.
 	 */
 	public static String guessMavenHome() {
-		String mvnHome = null;
-		try {
-			String[] cmd;
-			if (System.getProperty("os.name").contains("Windows")) {
-				cmd = new String[]{"mvn.cmd", "-version"};
-			} else if (System.getProperty("os.name").contains("Mac")) {
-				cmd = new String[]{"sh", "-c", "mvn -version"};
-			} else {
-				cmd = new String[]{"mvn", "-version"};
-			}
-			Process p = Runtime.getRuntime().exec(cmd);
-			try (BufferedReader output = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
-				String line;
-
-				while ((line = output.readLine()) != null) {
-					if (line.contains(mavenVersionParsing)) {
-						return line.replace(mavenVersionParsing, "");
-					}
-				}
+		String executableName;
+		if (System.getProperty("os.name").contains("Windows")) {
+			executableName = "mvn.cmd";
+		} else {
+			executableName = "mvn";
+		}
+		for (String dirname : System.getenv("PATH").split(File.pathSeparator)) {
+			File file = new File(dirname, executableName);
+			if (file.isFile() && file.canExecute()) {
+				return file.getAbsolutePath();
```

Comment:
This doesn't seem right to me. "Maven home" is the path to the Maven install directory, not the path to the Maven executable.

Does this actually work for running the `MavenLauncher`?

I would also advice extracting this into a separate method ('Path getExecutablePath(String name)`) or something like that, and throw the exception from there instead.

Reply:
@slarse `Maven home` and `Maven executable` are indeed different things. I tried to look up Maven home in PATH, but it did not exist, so I checked how the path to maven home is being used in the codebase. I found out that there were only two places.
1. Used to [generate a classpath file](https://github.com/INRIA/spoon/blob/a9aa2296f4858baa6188955b9be63a849135d8de/src/main/java/spoon/support/compiler/SpoonPom.java#L571). I did not meticulously read the method definition, but it seems like it tries to build the classpath if it doesn't exist.
2. Used to [run checkstyle](https://github.com/INRIA/spoon/blob/a9aa2296f4858baa6188955b9be63a849135d8de/src/test/java/spoon/test/prettyprinter/DefaultPrettyPrinterTest.java#L525) in this test, although it is ignored.

From these two usages, I felt that looking up the path to executable would suffice, and maven home is not actually needed. 

> Does this actually work for running the MavenLauncher?

Yes, it does, and this corroborated my patch.

Reply:
I decided to keep the intended behaviour. Pushed the changes for it.